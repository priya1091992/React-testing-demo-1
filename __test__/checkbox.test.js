import React from 'react';
import { mount, shallow } from 'enzyme';
import CheckboxWithLabel from '../src/checkbox';
import renderer from 'react-test-renderer';

test('CheckboxWithLabel changes the text after click', () => {
  const checkbox = mount(<CheckboxWithLabel labelOn="On" labelOff="Off" />);
  expect(checkbox.text()).toEqual('Offwww.gmail.com');
  checkbox.find('input').simulate('change');
  expect(checkbox.text()).toEqual('Onwww.gmail.com');
  expect(checkbox.contains(<a href="www.gmail.com">www.gmail.com</a>)).toEqual(true);
});

test('CheckboxWithLabel', () => {
  const wrapper = renderer.create(
    <CheckboxWithLabel labelOn="On" labelOff="Off" />
  );
  let tree = wrapper.toJSON();
  expect(tree).toMatchSnapshot();
});
