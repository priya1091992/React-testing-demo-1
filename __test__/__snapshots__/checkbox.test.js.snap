// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Off
  <a
    href="www.gmail.com"
  >
    www.gmail.com
  </a>
</label>
`;
